membrane_curvature/tests/test_mdakit_membcurv.py

```diff
@@ -144,44 +151,152 @@ def test_mean_curvature_all():
+    # number of cells
+    3,
+    # sum of z coordinate in grid
+    np.array([[10., 10., 10.],
+              [10., 10., 10.],
+              [10., 10., 10.]]),
+    # grid number of beads per unit
+    np.array([[2., 1., 1.],
+              [1., 2., 1.],
+              [1., 1., 2.]]),
+    # avg z coordinate in grid
+    np.array([[5., 10., 10.],
+              [10., 5., 10.],
+              [10., 10., 5.]]),
+)])
+def test_avg_unit_cell_more_beads(n_cells, grid_z_coords, grid_norm, grid_avg):
+    z_ref = np.zeros([n_cells, n_cells])
+    unit_cell = avg_unit_cell(z_ref, grid_z_coords, grid_norm)
+    for cell in range(n_cells):
+        assert_almost_equal(unit_cell[cell], grid_avg[cell])
+
+
+@pytest.mark.parametrize('n_cells, max_width, z_avg', [(
+    # number of cells
+    # max_width is 3nm/30A (from GRO_PO4_SMALL)
+    3, 30.,
+    # z ref of surface
+    # z coordinate in grid
+    np.array(([150., 150., 120.],
+              [150., 120., 120.],
+              [150., 120., 120.])))])
+def test_derive_surface(small_grofile, n_cells, max_width, z_avg):
+    max_width_x = max_width_y = max_width
+    surf = derive_surface(n_cells, small_grofile, max_width_x, max_width_y)
+    for cell, calc_cell in zip(z_avg, surf):
+        assert_almost_equal(cell, calc_cell)
+
+
+@pytest.mark.parametrize('n_cells, max_width', [(3, 300.)])
+@pytest.mark.parametrize('dummy_array, z_avg', [(
+    # dummy coordinates [x,y,z]
+    np.array([[0., 0., 150.],
+              [100., 0., 150.],
+              [200., 0., 150.],
+              [0., 100., 150.],
+              [100., 100., 120.],
+              [200., 100., 120.],
+              [0., 200., 120.],
+              [100., 200., 120.],
+              [200., 200., 120.]]),
+    # z_avg in grid
+    np.array(([150., 150., 120.],
+              [150., 120., 120.],
+              [150., 120., 120.]))
+)])
+def test_derive_surface_from_numpy(dummy_array, n_cells, max_width, z_avg):
+    u = mda.Universe(dummy_array, n_atoms=len(dummy_array))
+    selection = u.select_atoms('index 0:9')
+    max_width_x = max_width_y = max_width
+    surf = derive_surface(n_cells, selection, max_width_x, max_width_y)
+    for cell, calc_cell in zip(z_avg, surf):
+        assert_almost_equal(cell, calc_cell)
+
+
+@pytest.mark.parametrize('n_cells, max_width', [(3, 300.)])
+@pytest.mark.parametrize('dummy_array, z_avg', [(
+    # dummy coordinates [x,y,z]
+    np.array([[0., 0., 150.],
+              [0., 0., 150.],
+              [200., 0., 150.],
+              [0., 0., 150.],
+              [100., 100., 150.],
+              [200., 100., 150.],
+              [0., 200., 150.],
+              [100., 200., 150.],
+              [200., 200., 150.]]),
+    # z_avg in grid
+    np.array(([150., np.nan, 150.],
+              [np.nan, 150., 150.],
+              [150., 150., 150.]))
+)])
+def test_derive_surface_from_numpy_with_nans(dummy_array, n_cells, max_width, z_avg):
```

Comment:
Could you please delete this test, and it's now just a duplicate of test_get_z_surface?